Assets/HotmaxEditorVR/propertiesMenu/propertiesMenu.cs

```diff
@@ -0,0 +1,262 @@
+                }
+            }
+
+            if (hit.collider.gameObject.name == "materialPageDown")
+            {
+                if (currentPage <= 0)
+                {
+                    return;
+                }
+                else
+                {
+                    currentPage -= 1;
+                    removeLastMaterialPage();
+                    showMaterialPage(currentPage);
+                }
+            }
+
+            //--SETTING (just updating the UI for now)
+            if (hit.collider.gameObject.name == "toggleHasLocalAnimator")
+            {
+                if (hasLocalAnimator)
+                {
+                    //TODO: remove animator
+                    animatorText.text = "#no-animation";
+                    hasLocalAnimator = false;
+                }
+                else
+                {
+                    //TODO: Add animation controller class
+                    animatorText.text = "#has-animation";
+                    hasLocalAnimator = true;
+                }
+            }           
+        }
+    }
+
+    void triggerUnclicked(object sender, ClickedEventArgs e)
+    {
+        triggerDown = false;
+
+        if (sliderMoving)
+        {
+            if (sliderUnit < 6)
+            {
+                if (!_selectedObject.GetComponent<Rigidbody>())
+                {
+                    _selectedObject.AddComponent<Rigidbody>();
+                }
+                _selectedObject.GetComponent<Rigidbody>().mass = sliderUnit * 10;
+                _selectedObject.GetComponent<Rigidbody>().isKinematic = true;
+                physicsText.text = "#grabbable #throwable";
+
+            }
+            else if (sliderUnit > 6 && sliderUnit < 9)
+            {
+                if (!_selectedObject.GetComponent<Rigidbody>())
+                {
+                    _selectedObject.AddComponent<Rigidbody>();
+                    selectedObjectRb = _selectedObject.GetComponent<Rigidbody>();
+                }
+                _selectedObject.GetComponent<Rigidbody>().mass = sliderUnit * 10;
+                _selectedObject.GetComponent<Rigidbody>().isKinematic = true;
+                physicsText.text = "#heavy #non-grabbable";
+            }
+            else
+            {
+                if (_selectedObject.GetComponent<Rigidbody>())
+                {
+                    Destroy(_selectedObject.GetComponent<Rigidbody>());
+                }
+                physicsText.text = "#non-moveable";
+            }
+            sliderMoving = false;
+        }
+    }
+
+    void Update()
+    {
+        if (triggerDown)
+        {
+            RaycastHit hit;
+            Ray ray = new Ray(inputManager.hand2.gameObject.transform.position, inputManager.hand2.gameObject.transform.forward);
+            if (Physics.Raycast(ray, out hit, 100))
+            {
+                if (hit.collider.gameObject == slider.transform.parent.transform.parent.gameObject)
```

Comment:
same here, move to iHittable

Reply:
you could extrapolate out slider and "toggle" buttons basic functionality (a slider moves, a button is clicked on or off) to their own classes -- and have complex functionality reference public methods here in this class

Reply:
the slider needs to follow the raycast hot.point. I am not sure how to handle that with iHittable. Got any ideas?

Reply:
ok, made a fix to the slider.